### PR TITLE
Add quality signals (outcome inference + one-shot rate) (closes #6)

### LINKED ISSUE #6: Design: outcome / quality signal for 'same output, less spend' comparisons
## This is a design issue, not an implementation ticket.

Before building, we need a decision.

## Problem

Burn's meta-goal is answering \"would the same work have cost less with a different model / harness / tool choice?\" Spend data alone can't answer this — we also need a notion of whether the cheaper alternative produced an *equivalent* result. Otherwise \"Haiku would have cost 5× less\" is useless if Haiku would have produced broken code.

This is the moat versus every other usage tracker (TokenTracker, ccusage, Cursor stats). None of them have a quality axis. Whichever signal we pick, the data model must accommodate it from the start so it doesn't become a v2 retrofit.

## Candidate signals

**(a) Explicit outcome tag at session end**

```
burn tag --outcome good|bad|meh [--note \"…\"]
burn claude --tag workflow=refactor -- <claude args>
  # on exit, prompts: \"how did this go? [good/bad/meh/skip]\"
```

- Pros: simplest, most reliable when present, user controls the semantics.
- Cons: relies on discipline. Realistic capture rate probably 20–40% without prompts. Even with prompts, users will skip during flow.

**(b) Git-state scrape**

Record `git HEAD`, working-tree status, and test-suite state (`npm test`, `pytest`, etc. — whatever's there) at session start and end. Later: check whether the session's commits were reverted within N days, whether a PR containing them merged, whether tests still pass on HEAD.

- Pros: automatic. Captures the real-world \"did this stick\" signal.
- Cons: test harness detection is language/project-specific. Revert-within-N-days requires a delayed check (cron or on-demand). Won't capture quality for non-code tasks.

**(c) Benchmark harness**

Out of scope for v0.1 but worth considering in the data model. Re-run the same prompt across models with automated scoring. Requires storing prompts — which we explicitly don't do today for privacy.

**(d) Do nothing**

Report cost and let the user compare quality by hand. Simplest, ships fastest, but forfeits the differentiator.

## Hard constraint

Burn does not store prompts or responses — privacy is a deliberate design choice. Whatever signal we pick must not require storing them. This rules out (c) in its naive form; any benchmarking would need explicit opt-in with its own storage story.

## Asks of this issue

1. **Pick one** (or a hybrid — (a)+(b) is the obvious pairing).
2. Shape the data model now so `TurnRecord` / the ledger can carry an outcome without a schema migration later. Candidate: a sidecar `outcomes.jsonl` indexed by `(sessionId, workflowId?, projectKey)` with `{ signal, value, capturedAt, confidence? }`.
3. File implementation sub-issue(s).

## Decision criteria

- How many real sessions will actually produce an outcome signal? (If (a) alone captures < 20%, it's probably not worth the UX cost.)
- Does the signal survive being wrong? (Test suites lie; user tags are subjective; revert rate has false positives.)
- How much does it cost to ignore this for now and bolt on later?

Meta: the cheapest path may be (a) as the MVP with (b) as a later automatic augment — but worth arguing through before committing.

### PR DESCRIPTION
**Pull request by willwashburn**
## Summary

- Resolves issue #6 (outcome/quality signal design) by adopting the two signals the discussion converged on: **outcome inference** (agentsview) and **one-shot rate** (codeburn). They're orthogonal — each catches a different failure mode — and together form the MVP quality axis that separates \"cheaper model would have worked\" from \"cheaper model would have broken everything.\"
- Implements both in `packages/analyze/src/quality.ts`, wires into `burn summary --quality` as the first consumer.

**Stacked on [#52](https://github.com/AgentWorkforce/burn/pull/52)** — this PR uses `ToolCall.isError` (added in #52) for trailing-failure-streak detection. Merge #52 first, then this rebases onto main cleanly.

Closes #6.

### Design decision (the actual resolution of issue #6)

After the discussion on the issue, the candidate ranking settled on:

- (a) Explicit tagging — optional enhancement, not primary
- (b) Git-state scrape — future option
- (c) Benchmark harness — out of scope
- (d) Do nothing — rejected (forfeits the differentiator)
- (e) Automatic adherence detection (prism) — rejected (doesn't scale beyond hand-coded per-rule checkers)
- **(f) Outcome inference (agentsview) — primary**
- **(g) One-shot rate (codeburn) — primary**

(f) + (g) is the MVP. This PR lands both as concrete implementations, which closes the design issue.

### Why this pair

| Signal | Captures | Needs content? | Granularity |
|---|---|---|---|
| Outcome inference | abandoned / give-up sessions | Optional (degrades gracefully) | Per-session |
| One-shot rate | retry-heavy / stuck sessions | No | Per-turn → per-session |

The second row matters: a session can be `completed` with a low one-shot rate, which is *exactly* the \"Sonnet nailed it first try\" vs \"Haiku got there after 3 retries\" distinction that drives model-choice decisions. Neither signal alone catches this; both together do.

### Hard constraints honored

- No prompt storage required. Outcome inference works from turn metadata; last-assistant-text is used only to downgrade confidence, never required.
- Nothing persists to the ledger — both signals are computed lazily at query time so upgrading the rules later doesn't require a rebuild.
- Confidence is explicit on every outcome classification so downstream consumers can filter low-confidence signals rather than treat them as noise.

### What's not in this PR

- Explicit user-tagging UI (`burn tag --outcome`). Design demoted it to optional enhancement; worth a follow-up issue once the auto-classifier has real usage data.
- Git-state scrape (b). Future option, separate PR if demand warrants.

## Test plan

- [x] `pnpm run test:ts` — 238 pass (13 new quality tests covering each outcome reason + one-shot edge cases)
- [x] `pnpm dev:cli summary --quality --since 7d` against real sessions — outputs sensible rollup (51 completed / 8 abandoned / 0 errored / 10 unknown, 99.7% one-shot rate across 1265 edit turns in my personal ledger)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/agentworkforce/burn/pull/53" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

**Review by barryollama — APPROVED** [review]

File: packages/analyze/src/quality.ts [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+export interface OneShotMetrics {
+  sessionId: string;
+  editTurns: number;
+  oneShotTurns: number;
+  // `oneShotTurns / editTurns` when editTurns > 0, else undefined. Callers
+  // decide what to display for zero-edit sessions (NaN vs "—").
+  oneShotRate: number | undefined;
+  // Total retries across all turns in the session. Useful alongside the rate
+  // as a raw volume signal — a rate of 0.5 with 2 edits reads very different
+  // from a rate of 0.5 with 40 edits.
```
Comment (barryollama) [inline packages/analyze/src/quality.ts:49]: **Nit:** Consider adding "I cannot access", "unable to verify", and "doesn't appear to exist" to the patterns. These are common Claude/Code give-up phrases I've observed.

**Review by barryollama — COMMENTED** [review]
LGTM overall. Clean implementation of the two orthogonal signals from #6. One minor suggestion inline.
